temporal-sdk/src/main/java/io/temporal/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -196,7 +190,7 @@ public Result handle(ActivityTask activityTask, Scope metricsScope, boolean loca
     ActivityTaskExecutor activity = activities.get(activityType);
     if (activity == null) {
       if (dynamicActivity != null) {
-        return dynamicActivity.execute(activityInfo, metricsScope);
+        return dynamicActivity.execute(activityInfo, metricsScope, activityTask.getIdentity());
```

Comment:
Looking at the structure of this code, `activityTask.getIdentity()` should go inside `ActivityInfoInternal`

Reply:
The structure of the data flow is strange. activityTask should have an information about the task coming from the "outside". Worker identity is an internal information of the worker and should be coming from "inside".
I think that identity should be set in a constructor of POJOActivityTaskHandler and be provided by worker initialization code. Like it's done with `interceptors` or `dataConverter`.

Reply:
As noted in the first comment, it's still unclear why identity deserved it's own place inside
```
    ActivityTaskHandler.Result execute(
        ActivityInfoInternal task, Scope metricsScope, String identity);
```

Actually, it doesn't need to be passed at all. See how `POJOActivityImplementation`accesses `namespace` - it can and should access identity the same way, because why not.